ci-operator/templates/openshift/openshift-azure/build-azure-image.yaml

```diff
@@ -115,12 +117,8 @@ objects:
         echo "RESOURCE_GROUP=${RESOURCE_GROUP}"
         echo "IMAGE_NAME=${IMAGE_NAME}"
 
-        # The ansible playbook expects to source a file named "credentials"
-        cp /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/secret /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/credentials
-        # TODO: Remove once the following change in cherrypicked in release-3.10
-        # https://github.com/openshift/openshift-ansible/pull/10118
-        sed -i -e "s#AZURE_CLIENT_SECRET#AZURE_SECRET#" /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/credentials
-        sed -i -e "s#AZURE_TENANT_ID#AZURE_TENANT#" /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/credentials
+        source /etc/azure/credentials/secret
+        az login --service-principal -u ${AZURE_CLIENT_ID} -p ${AZURE_CLIENT_SECRET} --tenant ${AZURE_TENANT_ID} &>/dev/null
```

Comment:
@kwoodson here is where I am stuck: since we cannot hijack the entrypoint during runtime, I move `az login` (which is what the entrypoint is running too) before running the playbook here. This gets us past the [first task](https://github.com/openshift/openshift-ansible/blob/0d33b828698630a7b084df07cc4b63df4e5ac008/playbooks/azure/openshift-cluster/build_node_image.yml#L6) in the playbook where `az` is used but fails to [create a resource group](https://github.com/openshift/openshift-ansible/blob/0d33b828698630a7b084df07cc4b63df4e5ac008/playbooks/azure/openshift-cluster/build_node_image.yml#L10) subsequently. The [RG creation task](https://github.com/openshift/openshift-ansible/blob/0d33b828698630a7b084df07cc4b63df4e5ac008/playbooks/azure/openshift-cluster/tasks/provision_instance.yml#L3) talks to Azure via ansible and not via `az`. Unclear why it is not working though (it works in Jenkins, right?)
```
2018/09/18 10:34:51 Running pod build-azure-image
No service account file found at /usr/share/ansible/openshift-ansible/inventory/dynamic/injected/credentials, bypassing login
RESOURCE_GROUP=ci-op-2zfnzjw2-c883c-201809181034
IMAGE_NAME=centos7-3.10-201809181034

PLAY [localhost] ***************************************************************

TASK [calculate input image] ***************************************************
Tuesday 18 September 2018  10:34:58 +0000 (0:00:00.143)       0:00:00.143 ***** 
changed: [localhost]

TASK [create resource group] ***************************************************
Tuesday 18 September 2018  10:35:00 +0000 (0:00:02.513)       0:00:02.657 ***** 
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed to authenticate with provided credentials. Some attributes were missing. Credentials must include client_id, secret and tenant or ad_user and password, or ad_user, password, client_id, tenant and adfs_authority_url(optional) for ADFS authentication, or be logged in using AzureCLI."}
```
https://storage.googleapis.com/origin-ci-test/logs/azure-build-node-image-rhel-310/21/build-log.txt
Any ideas how to proceed?

Reply:
May it be that I need to source `AZURE_SECRET` and `AZURE_TENANT` instead of `AZURE_CLIENT_SECRET` and `AZURE_TENANT_ID`? That's the only difference with what we are doing in master.

Reply:
> May it be that I need to source AZURE_SECRET and AZURE_TENANT instead of AZURE_CLIENT_SECRET and AZURE_TENANT_ID? That's the only difference with what we are doing in master.

Yes, it turns out we need those specific variables :/